app/javascript/components/tree-view/reducers/oncheck.js

```diff
@@ -0,0 +1,90 @@
+import Tree from '@manageiq/react-ui-components/dist/wooden-tree';
+
+const getSelectedKeys = state => Object.values(state).filter(entry => entry.state && entry.state.checked).map(entry => entry.attr.key);
+
+const checkUrl = (node, type = '') => {
+  switch (type) {
```

Comment:
This is very 🦛🦛 at least we should have a FIXME to move these out from here.

Reply:
Also the case syntax is everything but readable.

Reply:
I could replace them by functions of their own? I would like to keep the grouped in one place so one can see the high diversity of the URL formats.

Reply:
I am for readability here, and in other switch statements too